pytket/pytket/circuit/add_condition.py

```diff
@@ -36,24 +36,23 @@
 
 
 def _add_condition(
-    circ: Circuit, condition: Union[ConstPredicate, Bit, BitLogicExp]
+    circ: Circuit, condition: Union[RegPredicate, Bit, BitLogicExp]
 ) -> Tuple[Bit, bool]:
     """Add a condition expression to a circuit using classical expression boxes,
     rangepredicates and conditionals. Return predicate bit and value of said bit.
     """
     if isinstance(condition, Bit):
         return condition, True
-    elif isinstance(condition, ConstPredicate):
+    elif isinstance(condition, RegPredicate):
         pred_exp, pred_val = condition.args
-        # ConstPredicate constructor should ensure arg order
+        # RegPredicate constructor should ensure arg order
```

Comment:
this comment is no longer true, you will need to add an exception here if the `pred_val` is not a constant (or handle it by adding the expression directly)

Reply:
hm, shouldn't we simply remove the assertion ? Or, do you mean that further down th e`bool(pred_val )` will fail ?

Reply:
yes exactly, the rest of this function only applies if the value is a bool, so we either need to raise an exception or handle it, for now I think raising an exception is the right thing to do (instead of the assert).

Reply:
Oh, I see. I was confused because now, `pred_val` can be something else than a `Constant` i.e. a register and I didn't see how you could get a bool from it. But what you mean is that this `add_condition` method must work identically to what it was defined for: with the second argument being a `Constant` despite the extension of types for the predicate.

Reply:
Done.

Reply:
`NotImplemented` is not very descriptive, ideally use a custom exception like `NonConstError` with a helpful message like "Condition expressions must be Predicates which have a constant as their second operand".

Reply:
Sure thing. Done.